### DISCUSSION TITLE
How to check if queried record doesn't exist?

### DISCUSSION THREAD

**Discussion by intervinn** [go-gorm/gorm#6084] [Q&A] [answered]
I am quite new to GORM and SQL databases, so far, this is not working code I've got:
```go
if lib.DB.First(&models.User{Email: payload.Email}) == nil {
```
So, how do I check if the queried record doesn't exist?

**Comment by acynothia** [comment, marked as answer]
```
if err := if lib.DB.First(&models.User{Email: payload.Email}).Error; errors.Is(err, gorm.ErrRecordNotFound) {
```

**Reply by phantomdenkali** [reply]
I'm running into the same error, though I think I'm using what you suggested. Is there an issue with the versions I am using?

https://github.com/go-gorm/gorm/discussions/6485

**Reply by phantomdenkali** [reply]
never mind, my database was locked and the query was failing.
